packages/blocks/src/api/validation.js

```diff
@@ -397,6 +397,32 @@ function getHTMLTokens( html ) {
 	return null;
 }
 
+/**
+ * Returns true if the next HTML tag closes the current tag, false otherwise. Note this check does not
+ * affect the array of tokens
+ *
+ * @param {Object} currentToken Current token to compare against.
+ * @param {Array} tokens Array of remaining HTML tokens.
+ *
+ * @return {boolean} Whether the next token in `tokens` closes `currentToken`, false otherwise
+ */
+function isClosedByNextToken( currentToken, tokens ) {
+	// Check if this is a self closing tag
+	if ( ! currentToken.selfClosing ) {
+		return false;
+	}
+
+	// Peek at the next token without consuming
+	const [ next ] = tokens;
```

Comment:
Nit: Consistency: We otherwise use "Token" as a suffix, but here we don't assign as `nextToken`